### PR TITLE
Expose parts for styling

### PR DESCRIPTION
**Pull request by DiegoCardoso**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-crud/182)

### CODE REVIEW

File: src/vaadin-crud.html [hunk 1]
```diff
@@ -60,8 +60,8 @@
       }
     </style>
 
-    <div id="container">
-      <div id="main">
+    <div id="container" part="container">
+      <div id="main" part="main">
```
Comment (DiegoCardoso) [inline src/vaadin-crud.html]: I am not fully happy with the names, but couldn't find better ones.

Comment (yuriy-fix) [reply src/vaadin-crud.html]: Not sure, but I guess there is no need in exposing `main` in that case as you most probably will get access to it via `[part="container"] > div`.

**Comment by DiegoCardoso** [comment]
> Is there any reason why those styling parts are needed to be exposed? 🤔

Most straight forward example is Bakery, where the toolbar is placed on top, by changing the `flex-direction` to `row-reverse`. As the grid and toolbar were straight under the host element, was just a matter of styling that with:
```css
:host {
  flex-direction: column-reverse;
}
```

Since now it's hidden on some internal structure, I believe it makes sense to expose some parts.

File: src/vaadin-crud.html (hunk 1 above)
Comment (DiegoCardoso) [reply src/vaadin-crud.html]: Well, true, I thought it would more safe to expose more parts, but if `[part="container] > div` is ok, then it may be fine.

**Comment by yuriy-fix** [comment]
___
*[src/vaadin-crud.html, line 64 at r1](https://reviewable.io/reviews/vaadin/vaadin-crud/182#-LsMl8Zo_rbsixk7EYll-r1-64:-LsaGBUW28paOG3aiQti:b-xuidqi) ([raw file](https://github.com/vaadin/vaadin-crud/blob/166541dfd4a1f501f6b33d8f36a9126c6bf975ef/src/vaadin-crud.html#L64)):*
<details><summary><i>Previously, DiegoCardoso (Diego Cardoso) wrote…</i></summary><blockquote>

Well, true, I thought it would more safe to expose more parts, but if `[part="container] > div` is ok, then it may be fine.
</blockquote></details>

Hmm, we can ask for 3rd opinion here as well as I don't have a strong one.
@web-padawan

File: src/vaadin-crud.html (hunk 1 above)
Comment (jouni) [reply src/vaadin-crud.html]: `[part="container"] > div` is an unsupported selector, so we should not encourage anyone to use those.

I don’t see any semantic purpose for exposing those elements for styling. They look very much just like internal layout implementation details, which the developer should have no interest in styling separately.

**Comment by jouni** [comment]
I agree with Yuriy and Serhii. No need to expose those elements. The use case of moving the toolbar above the grid is something we might want to support more explicitly. But since CSS `order` can be used to workaround that without introducing more stylable parts or using unsupported selectors, I think that’s fine for now.

I mean, we might in the future change the internal layout implementation so that `order` no longer does what you expect, and then we should probably acknowledge that and support that use case in some other way. Alternatively, we should stop promoting that use case for now, until we have more explicit support.

**Comment by DiegoCardoso** [comment]
Alright. Thanks for the insights. I will close the PR, then.

**Pull request closed without merging**